lib/core/src/sdk.rs

```diff
@@ -1924,6 +1999,76 @@ impl LiquidSdk {
         })
     }
 
+    /// Performs a Liquid send payment via SideSwap
+    async fn pay_sideswap(
+        &self,
+        sideswap_service: &SideSwapService,
+        address_data: LiquidAddressData,
+        receiver_amount_sat: u64,
+        fees_sat: u64,
+    ) -> Result<SendPaymentResponse, PaymentError> {
+        let receiver_address =
+            elements::Address::from_str(&address_data.address).map_err(|err| {
+                PaymentError::generic(format!("Could not convert destination address: {err}"))
+            })?;
+        let asset_id = address_data.asset_id.clone().context("Expected asset id")?;
+        let swap = sideswap_service
+            .get_asset_swap(AssetId::from_str(&asset_id)?, receiver_amount_sat)
+            .await?;
+
+        ensure_sdk!(
+            swap.fees_sat <= fees_sat,
+            PaymentError::InvalidOrExpiredFees
+        );
+
+        ensure_sdk!(
+            self.get_info().await?.wallet_info.balance_sat >= swap.payer_amount_sat,
+            PaymentError::InsufficientFunds
+        );
+
+        let tx_id = sideswap_service
+            .execute_swap(receiver_address, &swap)
+            .await?;
+
+        // We insert a pseudo-tx in case LWK fails to pick up the new mempool tx for a while
+        // This makes the tx known to the SDK (get_info, list_payments) instantly
+        self.persister.insert_or_update_payment(
+            PaymentTxData {
+                tx_id: tx_id.clone(),
+                timestamp: Some(utils::now()),
+                fees_sat: swap.fees_sat,
+                is_confirmed: false,
+                unblinding_data: None,
+            },
+            &[
+                PaymentTxBalance {
+                    asset_id: utils::lbtc_asset_id(self.config.network).to_string(),
+                    amount: swap.payer_amount_sat,
+                    payment_type: PaymentType::Send,
+                },
+                PaymentTxBalance {
+                    asset_id: asset_id.to_string(),
+                    amount: swap.receiver_amount_sat,
+                    payment_type: PaymentType::Receive,
+                },
+            ],
```

Comment:
I'm a bit lost here. Why do we assume we are also receiving the payment? The interface suggests it can be sent to any address...